src/TypeAdapter/Factory/WrappedTypeAdapterFactory.php

```diff
@@ -48,30 +48,21 @@ public function __construct(TypeAdapter $typeAdapter, TypeToken $type, bool $str
         $this->strict = $strict;
     }
 
-    /**
-     * Will be called before ::create() is called.  The current type will be passed
-     * in.  Return false if ::create() should not be called.
-     *
-     * @param TypeToken $type
-     * @return bool
-     */
-    public function supports(TypeToken $type): bool
-    {
-        return $this->strict
-            ? $type->getRawType() === $this->type->getRawType()
-            : $type->isA($this->type->getRawType());
-    }
-
     /**
      * Accepts the current type and a [@see TypeAdapterProvider] in case another type adapter needs
-     * to be fetched during creation.  Should return a new instance of the TypeAdapter.
+     * to be fetched during creation.  Should return a new instance of the TypeAdapter. Will return
+     * null if the type adapter is not supported for the provided type.
      *
      * @param TypeToken $type
      * @param TypeAdapterProvider $typeAdapterProvider
-     * @return TypeAdapter
+     * @return TypeAdapter|null
      */
-    public function create(TypeToken $type, TypeAdapterProvider $typeAdapterProvider): TypeAdapter
+    public function create(TypeToken $type, TypeAdapterProvider $typeAdapterProvider): ?TypeAdapter
     {
-        return $this->typeAdapter;
+        if ($this->strict) {
+            return $type->rawType === $this->type->rawType ? $this->typeAdapter : null;
+        }
+
+        return $type->isA($this->type->rawType) ? $this->typeAdapter : null;
```

Comment:
Why not modify the `isA` method to be like this:

```php
/**
 * Returns true if the type matches the class, parent, full type, or one of the interfaces
 *
 * @param string $type
 * @param bool $strict
 * @return bool
 */
public function isA(string $type, bool $strict = false): bool
{
    if ($this->rawType === $type) {
        return true;
    }

    if ($this->fullTypeString === $type && !$strict) {
        return true;
    }

    if (\in_array($type, $this->parents, true) && !$strict) {
        return true;
    }

    return false;
}

```

Then this create method could be simplified to:

```php
/**
 * Accepts the current type and a [@see TypeAdapterProvider] in case another type adapter needs
 * to be fetched during creation.  Should return a new instance of the TypeAdapter. Will return
 * null if the type adapter is not supported for the provided type.
 *
 * @param TypeToken $type
 * @param TypeAdapterProvider $typeAdapterProvider
 * @return TypeAdapter|null
 */
public function create(TypeToken $type, TypeAdapterProvider $typeAdapterProvider): ?TypeAdapter
{
    return $type->isA($this->type->rawType, $this->strict) ? $this->typeAdapter : null;
}
```

Reply:
I don't want to have to pay for it if it's not required.